src/lib/y2network/sysconfig/connection_config_readers/base.rb

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/connection_config/ip_config"
+require "y2network/ip_address"
+
+module Y2Network
+  module Sysconfig
+    module ConnectionConfigReaders
+      # This is the base class for connection config readers.
+      #
+      # The derived classes should implement {#update_connection_config} method.
+      # methods.
+      class Base
+        # @return [Y2Network::Sysconfig::InterfaceFile] Interface's configuration file
+        attr_reader :file
+
+        # Constructor
+        #
+        # @param file [Y2Network::Sysconfig::InterfaceFile] Interface's configuration file
+        def initialize(file)
+          @file = file
+        end
+
+        # Builds a connection configuration object
+        #
+        # @return [Y2Network::ConnectionConfig::Base]
+        def connection_config
+          connection_class.new.tap do |conn|
+            conn.bootproto = BootProtocol.from_name(file.bootproto || "static")
+            conn.description = file.name
+            conn.interface = file.interface
+            conn.ip_configs = ip_configs
+            conn.startmode = Startmode.create(file.startmode || "manual")
+            conn.startmode.priority = file.ifplugd_priority if conn.startmode.name == "ifplugd"
+            update_connection_config(conn)
+          end
+        end
+
+      private
+
+        # Returns the class of the connection configuration
+        #
+        # @return [Class]
+        def connection_class
+          self.class.to_s[/::([^:]+)$/, 1]
```

Comment:
```suggestion
          class_name = self.class.to_s[/::([^:]+)$/, 1]
```

Reply:
hmm and test passed, that is strange. It should be covered.

Reply:
I forgot to drop the old methods :man_facepalming: